db/view/view.cc

```diff
@@ -1289,6 +1320,13 @@ future<std::optional<utils::chunked_vector<frozen_mutation_and_schema>>> view_up
         co_await advance_existings();
     }
 
+    if (is_update_start && _update_partition_tombstone && _update->is_end_of_partition()) {
+        // The only update is the partition tombstone.
+        // For views that have the same partition key as base, generate an update of partition tombstone to delete
+        // the entire partition in one operation, instead of generating an update for each row.
+        generate_update(_update_partition_tombstone);
+    }
```

Comment:
Ah, it looks like this logic won't be triggered in the situation I described in the other comment (i.e. there is a partition tombstone and some clustered rows).

Perhaps such a situation should rather happen rarely, but given the performance impact of generating a large amount of view updates which happens when this optimization does not kick in, I think we should address it after all.

Reply:
Yes, I went on the safe side and considered the optimization when we have only a partition tombstone and no other updates.
I will try to think if I can relax this. Maybe let through all updates with a newer timestamp

Reply:
I changed it and removed the restriction of the partition tombstone being the only update.
Now in get_view_rows I skip generating the row updates if I see that we have a partition tombstone and the update is an equal row tombstone, so this should skip the individual row tombstones but allow other type of updates